app/controllers/clients_controller.rb

```diff
@@ -21,17 +21,8 @@ def show
     @client = Client.find(params[:id])
     @contact_history = (@client.outgoing_text_messages + @client.incoming_text_messages + @client.outgoing_emails
                        ).sort_by(&:datetime)
-  end
-
-  def send_text
-    outgoing_text_message = OutgoingTextMessage.create(
-      client: Client.find(params[:client_id]),
-      body: params[:body],
-      sent_at: DateTime.now,
-      user: current_user,
-    )
-    SendOutgoingTextMessageJob.perform_later(outgoing_text_message.id)
-    redirect_to client_path(id: params[:client_id])
+    @contact_history = (@client.outgoing_text_messages + @client.incoming_text_messages).sort_by(&:datetime)
```

Comment:
Seems duplicated with line 22. IMHO you should remove one definition of `@contact_history = ...`